common/src/main/java/org/conscrypt/NativeCrypto.java

```diff
@@ -52,13 +68,39 @@
 @Internal
 public final class NativeCrypto {
     // --- OpenSSL library initialization --------------------------------------
+    private static final Throwable UNAVAILABILITY_CAUSE;
     static {
-        NativeCryptoJni.init();
-        clinit();
+        Throwable cause = null;
```

Comment:
Is it intentional that previous failure cases will see a delayed failure?  Not against this, but also wondering if the plan is to call checkAvailability() at a few key points in another PR?